prowler/providers/azure/lib/service/service.py

```diff
@@ -20,23 +23,44 @@ def __init__(
 
         self.audit_config = audit_info.audit_config
 
-    def __set_clients__(self, identity, credentials, service, region_config):
+    def __set_clients__(self, identity, credentials, services, region_config):
         clients = {}
         try:
-            if "GraphServiceClient" in str(service):
-                clients.update({identity.domain: service(credentials=credentials)})
+            if "GraphServiceClient" in str(services):
```

Comment:
For me it's a little strange to keep two different versions of the `clients` object whether it requires the `GraphServiceClient` or not. Could we have a common version for all? Like:
```json
"clients" : {
      "tenant_domain":{
         "client_name_1":"client_object_1",
         "client_name_2":"client_object_2"
      }
}
```